cmd/dlt/service/cmd.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	msv1 "github.com/openshift-online/ocm-sdk-go/servicemgmt/v1"
+	"github.com/openshift/rosa/pkg/interactive/confirm"
+	"github.com/openshift/rosa/pkg/logging"
+	"github.com/openshift/rosa/pkg/ocm"
+	rprtr "github.com/openshift/rosa/pkg/reporter"
+)
+
+var args struct {
+	// ID of service
+	ID string
+}
+
+var Cmd = &cobra.Command{
+	Use:   "service",
+	Short: "Deletes a service",
+	Long:  "Deletes a service.",
+	Example: `  # Delete a service with ID "aabbcc"
+  rosa delete service --id=aabbcc`,
+	Run: run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	flags.StringVar(&args.ID,
+		"id",
+		"",
+		"The ID of the service to be deleted.")
+}
+
+func run(cmd *cobra.Command, _ []string) {
+	reporter := rprtr.CreateReporterOrExit()
+	logger := logging.CreateLoggerOrExit(reporter)
+
+	// Create the client for the OCM API:
+	ocmClient, err := ocm.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create OCM connection: %v", err)
+		os.Exit(1)
+	}
+	defer func() {
+		err = ocmClient.Close()
+		if err != nil {
+			reporter.Errorf("Failed to close OCM connection: %v", err)
+		}
+	}()
+
+	if !confirm.Confirm("delete service with id '%s'", args.ID) {
+		os.Exit(0)
+	}
+
+	// First get the service to report additional resources
+	// that must be manually deleted.
+	service, err := ocmClient.GetManagedService(ocm.DescribeManagedServiceArgs{ID: args.ID})
```

Comment:
What if we do not have the service in the database? (and err management.)

Reply:
We are leveraging managed service to do all of these checks for us. Managed service will return an error and we will report it to the end user.

Reply:
I think it makes sense to add basic error reporting here:
```
if err != nil {
  reporter.Errorf("%s", err)
  os.Exit(1)
}
```

Reply:
Oh! I didn't realize I didn't add any error handling to this call.